src/libNode/Node.cpp

```diff
@@ -243,6 +254,144 @@ void Node::AddGenesisInfo(SyncType syncType) {
   }
 }
 
+bool Node::ValidateDB() {
+  deque<pair<PubKey, Peer>> dsComm;
+  const string lookupIp = "127.0.0.1";
+  const uint port = 30303;
+
+  for (const auto& dsKey : *m_mediator.m_initialDSCommittee) {
+    dsComm.emplace_back(dsKey, Peer());
+  }
+  std::list<BlockLink> blocklinks;
+  if (!BlockStorage::GetBlockStorage().GetAllBlockLink(blocklinks)) {
+    LOG_GENERAL(WARNING, "BlockStorage skipped or incompleted");
+    return false;
+  }
+
+  blocklinks.sort([](const BlockLink& a, const BlockLink& b) {
+    return std::get<BlockLinkIndex::INDEX>(a) <
+           std::get<BlockLinkIndex::INDEX>(b);
+  });
+
+  std::list<TxBlockSharedPtr> txblocks;
+  if (!BlockStorage::GetBlockStorage().GetAllTxBlocks(txblocks)) {
+    LOG_GENERAL(WARNING, "Fail to get Tx Blocks");
+    return false;
+  }
+
+  txblocks.sort([](const TxBlockSharedPtr& a, const TxBlockSharedPtr& b) {
+    return a->GetHeader().GetBlockNum() < b->GetHeader().GetBlockNum();
+  });
+
+  const auto& latestTxBlockNum = txblocks.back()->GetHeader().GetBlockNum();
+
+  vector<boost::variant<DSBlock, VCBlock, FallbackBlockWShardingStructure>>
+      dirBlocks;
+  for (const auto& blocklink : blocklinks) {
+    if (get<BlockLinkIndex::BLOCKTYPE>(blocklink) == BlockType::DS) {
+      auto blockNum = get<BlockLinkIndex::DSINDEX>(blocklink);
+      if (blockNum == 0) {
+        continue;
+      }
+      DSBlockSharedPtr dsblock;
+      if (!BlockStorage::GetBlockStorage().GetDSBlock(blockNum, dsblock)) {
+        LOG_GENERAL(WARNING, "Could not rertv DS Block " << blockNum);
```

Comment:
"rertv" here, and "retrv" in other parts.  Maybe better to spell it out as "retrieve" so we don't have a hard time searching the logs.  Also, "Fail" and "Failed" both used.